packages/expo-file-system/ios/EXFileSystem/EXSessionTasks/EXSessionTaskDispatcher.m

```diff
@@ -0,0 +1,98 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+
+#import <EXFileSystem/EXSessionTaskDispatcher.h>
+#import <EXFileSystem/EXSessionResumableDownloadTaskDelegate.h>
+
+@interface EXSessionTaskDispatcher ()
+
+@property (nonatomic, strong) NSMutableDictionary<NSURLSessionTask *, EXSessionTaskDelegate *> *tasks;
+@property (nonatomic) BOOL isActive;
+
+@end
+
+@implementation EXSessionTaskDispatcher
+
+- (instancetype)init
+{
+  if (self = [super init]) {
+    _tasks = [NSMutableDictionary dictionary];
+    _isActive = true;
+  }
+  return self;
+}
+
+#pragma mark - public methods
+
+- (void)registerTaskDelegate:(EXSessionTaskDelegate *)delegate forTask:(NSURLSessionTask *)task
+{
+  _tasks[task] = delegate;
+}
+
+- (void)unregisterTaskDelegate:(NSURLSessionTask *)task
+{
+  EXSessionTaskDelegate *delegate = _tasks[task];
+  [self _unregisterIfResumableTaskDelegate:delegate];
+  [_tasks removeObjectForKey:task];
+}
+
+- (void)deactivate
+{
+  _isActive = false;
+}
+
+#pragma mark - helpers
+
+- (void)_unregisterIfResumableTaskDelegate:(EXSessionTaskDelegate *)taskDelegate
+{
+  if ([taskDelegate isKindOfClass:[EXSessionResumableDownloadTaskDelegate class]]) {
+    EXSessionResumableDownloadTaskDelegate *exResumableTask = (EXSessionResumableDownloadTaskDelegate *)taskDelegate;
+    [exResumableTask invalid];
```

Comment:
Could we move this to `EXSessionResumableDownloadTaskDelegate`? i.e. implement a `- (void)URLSession:downloadTask:didFinishDownloadingToURL:` callback which would remove the task from the manager?